File: Day<19>/Solution.cpp
#include <bits/stdc++.h>
#include <unordered_set>
using namespace std;

class Solution
{
    public:
    int candies(int m, int n)
    {
    	int X = (m * n) - m - n;
    	unordered_set<int> hash_set;
    	queue<int> queue;
    	queue.push(X);
    	hash_set.insert(X);

        int count = 0;
    	while (queue.size() > 0)
    	{
    		int curr = queue.front();
    		queue.pop();
    		count++;

    		int key = curr-m;
    		if (key > 0 && (hash_set.find(key) == hash_set.end()))
    		{
    			queue.push(key);
    			hash_set.insert(key);
    		}

    		key = curr-n;
    		if (key > 0 && (hash_set.find(key) == hash_set.end()))
    		{
    			queue.push(key);
    			hash_set.insert(key);
    		}
    	}
    	return count;
    }
};

int main() 
{ 
	int t;
	cin>>t;
	while (t--)
	{
		int m,n;
		cin>>m>>n;
		Solution obj;
		cout << obj.candies(m,n)<<endl;
	}
	return 0;
}

File: Day<19>/Solution.py
class Solution:
    def candies(self, m, n):
    	i = 0
    	X = (m * n) - m - n

    	queue = []
    	queue.append(X)
    	set = {X}

    	count = 0
    	while (len(queue) > 0):
    		curr = queue[0] 
    		queue.remove(queue[0])

    		count += 1
    		key = curr-m
    		if (key > 0 and key not in set):
    			queue.append(key)
    			set.add(key)

    		key = curr-n
    		if (key > 0 and key not in set):
    			queue.append(key)
    			set.add(key)
    	return count

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
    	arr = list(map(int, input().strip().split()))
    	m = arr[0]
    	n = arr[1]
    	obj = Solution()
    	print(obj.candies(m,n))
